graphics/wxWidgets-3.0/Portfile

```diff
@@ -272,11 +278,18 @@ if {${subport} eq "wxPython-3.0"} {
 
     configure.args-replace  --with-cocoa --with-gtk=3 \
                             --without-sdl --with-sdl
-}
 
-if {${subport} eq "wxgtk-3.0" || ${subport} eq "wxgtk-3.0-cxx11"} {
+    configure.cxxflags-append -std=c++11
+
+} elseif {${subport} eq "wxgtk-3.0"} {
+    # Address this bug: https://trac.macports.org/ticket/70124
+    patchfiles-append       patch-no-osx-evtloopsrc.h.diff
+    configure.cxxflags-append -std=c++03
+
+} elseif {${subport} eq "wxgtk-3.0-cxx11"} {
     # Address this bug: https://trac.macports.org/ticket/70124
     patchfiles-append       patch-no-osx-evtloopsrc.h.diff
+    configure.cxxflags-append -std=c++11
```

Comment:
1.  It builds perfectly fine with gcc14, which defaults to C++17. W/e is the problem with clang build, it is not the newer standard used for the build by the compiler.
2. Forcing C++11 here will require revbumping all ports which use wx*. Otherwise nothing gonna work, since wx* compares C++ standard at runtime (so you won’t see errors when building, but apps won’t launch). Assuming the intent is to have it actually working and not just nominally building.

Reply:
it **does** work, and nothing is broken by it that I have tried.

leaving it as it is, however, is broken on any system building with a compiler that defaults to c++17 (see all the 10.7 builds).

Reply:
it may still build with gcc-14… different compilers raise errors with different things at different times. But supporting clang-based builds is too important to leave it broken for that reason.

Reply:
Haha… the intent is always for things to be actually working, which is why I’m always after you to test your PRs on more than just 10,6-ppc.

Reply:
Well, maybe you use C++11 with Clangs and let GCC use whatever it wants by default, since unlike Clang, it builds this fine? Or leave GTK ports alone, others are meaningless on GCC-using systems anyway.

Reply:
No need for that, I believe. Too much spaghetti in the Portfiles just leads to needless confusion and pointless workarounds. Sorry, can't just leave the wxgtk ports alone, they are needed on older systems and are broken on some of them.